boot/zephyr/serial_adapter.c

```diff
@@ -115,7 +115,7 @@ boot_console_init(void)
 }
 
 static void
-boot_uart_fifo_callback(struct device *dev)
+boot_uart_fifo_callback(struct device *dev, void *user_data)
```

Comment:
It might be worth to mark `user_data` as `UNUSED` or `(void)user_data;` in code to avoid compiler warnings on unused variables (though I have not seen such warnings when compiling the change).